draft-ietf-suit-firmware-encryption.cddl

```diff
@@ -1,73 +1,4 @@
-; Define SUIT_Encryption_Info_* as a subset of COSE_Encrypt
-
-SUIT_Encryption_Info = #6.96(
-    SUIT_Encryption_Info_AESKW .within COSE_Encrypt /
-    SUIT_Encryption_Info_ESDH .within COSE_Encrypt)
-
-SUIT_Encryption_Info_AESKW = [
-  protected   : bstr .size 0 / bstr .cbor outer_header_map_protected,
-  unprotected : outer_header_map_unprotected,
-  ciphertext  : bstr / nil,
-  recipients  : [ + COSE_recipient_AESKW .within COSE_recipient ]
-]
-
-COSE_recipient_AESKW = [
-  protected   : bstr .size 0 / bstr .cbor empty_map,
-  unprotected : recipient_header_unpr_map_aeskw,
-  ciphertext  : bstr        ; CEK encrypted with KEK
-]
-empty_map = {}
-
-recipient_header_unpr_map_aeskw =
-{
-    1 => int,         ; algorithm identifier for non AEAD cipher
-  ? 4 => bstr,        ; identifier of the recipient public key
-  * label => values   ; extension point
-}
-
-SUIT_Encryption_Info_ESDH = [
-  protected   : bstr .size 0 / bstr .cbor outer_header_map_protected,
-  unprotected : outer_header_map_unprotected,
-  ciphertext  : bstr / nil,
-  recipients  : [ + COSE_recipient_ESDH .within COSE_recipient ]
-]
-
-COSE_recipient_ESDH = [
-  protected   : bstr .size 0 / bstr .cbor recipient_header_map_esdh,
-  unprotected : recipient_header_unpr_map_esdh,
-  ciphertext  : bstr        ; CEK encrypted with KEK
-]
-
-recipient_header_map_esdh =
-{
-  ? 1 => int,         ; algorithm identifier for AEAD cipher
-  * label => values   ; extension point
-}
-
-recipient_header_unpr_map_esdh =
-{
-   -1 => COSE_Key,    ; ephemeral public key for the sender
-  ? 1 => int,         ; algorithm identifier for non AEAD
-  ? 4 => bstr,        ; identifier of the recipient public key
-  * label => values   ; extension point
-}
-
-; common definitions
-outer_header_map_protected =
-{
-  ? 1 => int,         ; algorithm identifier for AEAD cipher
-  * label => values   ; extension point
-}
-
-outer_header_map_unprotected =
-{
-  ? 1 => int,         ; algorithm identifier for non AEAD cipher
-    5 => bstr,        ; IV
-  * label => values   ; extension point
-}
-
-
-; Extends SUIT Manifest
+SUIT_Encryption_Info = #6.96(COSE_Encrypt)
```

Comment:
To judge any tagged COSE_Encrypt binaries valid other than AES-KW and ECDH-ES+AES-KW.